File: src/main/java/com/github/nsc/de/compiler/parser/node/expression/IntegerNode.java
package com.github.nsc.de.compiler.parser.node.expression;

import com.github.nsc.de.compiler.parser.node.ValuedNode;

public class IntegerNode implements ValuedNode {

    private final int number;

    public IntegerNode(int number) {
        this.number = number;
    }

    public int getNumber() {
        return number;
    }

    @Override
    public String toString() {
        return String.valueOf(number);
    }
}

File: parser/src/commonMain/kotlin/com/github/shakelang/shake/parser/node/factor/IntegerNode.kt
package com.github.shakelang.shake.parser.node.factor

import com.github.shakelang.shake.parser.node.ValuedNode
import com.github.shakelang.shake.util.characterinput.position.PositionMap

class IntegerNode(map: PositionMap, val number: Int) : ValuedNode(map) {
    override fun toJson(): Map<String, *> = mapOf("name" to "IntegerNode", "value" to number)
}